pkg/v1/remote/transport/bearer.go

```diff
@@ -46,22 +46,37 @@ var _ http.RoundTripper = (*bearerTransport)(nil)
 
 // RoundTrip implements http.RoundTripper
 func (bt *bearerTransport) RoundTrip(in *http.Request) (*http.Response, error) {
-	hdr, err := bt.bearer.Authorization()
+	sendRequest := func() (*http.Response, error) {
+		hdr, err := bt.bearer.Authorization()
+		if err != nil {
+			return nil, err
+		}
+
+		// http.Client handles redirects at a layer above the http.RoundTripper
+		// abstraction, so to avoid forwarding Authorization headers to places
+		// we are redirected, only set it when the authorization header matches
+		// the registry with which we are interacting.
+		if in.Host == bt.registry.RegistryStr() {
```

Comment:
I haven't thought to do this in other situations where I'm injecting a `RoundTripper` -- maybe I should? Is there a specific case where you've seen this as a problem? Or is there some semi-official guidance that you should do this?

Maybe we should this change in a separate PR, just to be sure? And with tests, definitely.

Reply:
This was introduced in https://github.com/google/go-containerregistry/commit/29f7682fa6959d1c388ccac0f4261eb3e8b36d10.
The commit message mentions that some registries redirect to object storage that don't like the Authorization header.
`transport.basicTransport` has this as well: https://github.com/google/go-containerregistry/blob/master/pkg/v1/remote/transport/basic.go#L42

It looks like all `RoundTripper`'s in `pkg/v1/remote/transport` are covered.

Reply:
My mistake, I misread the diff and it looked like this was added in this PR.